src/renderer/components/cluster-manager/cluster-view.tsx

```diff
@@ -68,19 +94,43 @@ class NonInjectedClusterView extends React.Component<Dependencies> {
   bindEvents() {
     disposeOnUnmount(this, [
       reaction(() => this.clusterId, async (clusterId) => {
+        this.navigateToClusterDisposer?.();
+
+        if (!clusterId) {
+          return;
+        }
+
         this.props.clusterFrames.setVisibleCluster(clusterId);
         this.props.clusterFrames.initView(clusterId);
-        requestClusterActivation(clusterId, false); // activate and fetch cluster's state from main
+        this.props.requestClusterActivation({
+          clusterId,
+          force: false,
+        });
         this.props.entityRegistry.activeEntity = clusterId;
+
+        const navigateToClusterDisposer = disposer(
+          when(
+            () => this.cluster?.disconnected === false,
+            () => {
+              navigateToClusterDisposer.push(when(
+                // The clusterId check makes sure that we are still talking about the same cluster
+                () => (this.cluster?.disconnected ?? true) && this.clusterId === clusterId,
+                () => {
+                  this.props.navigateToCatalog();
+                },
+              ));
+            },
+          ),
+        );
+
+        this.navigateToClusterDisposer = navigateToClusterDisposer;
+
+        disposeOnUnmount(this, [
+          navigateToClusterDisposer,
+        ]);
```

Comment:
What is the thing that we are trying to do here? It's just so complex that it requires explanation. There must be simpler solution which doesn't require `reaction`, `when` or even `bindEvents` at all.